utils/json-utils/src/main/scala/pl/touk/nussknacker/engine/json/encode/BestEffortJsonSchemaEncoder.scala

```diff
@@ -96,6 +94,26 @@ object BestEffortJsonSchemaEncoder {
     }
   }
 
+  private def encodeIntegerSchema(value: Any, schema: Schema, fieldName: Option[String] = None): Validated[NonEmptyList[String], Json] = {
+    def encodeBigDecimalToIntegerSchema(bigDecimalValue: BigDecimal): Validated[NonEmptyList[String], Json] = Try(bigDecimalValue.toLongExact) match {
```

Comment:
Technically speaking, `.toLongExact` is too strict, as it rejects really big numbers, which would validate against integer schema (which doesn't have any max value). `.setScale(0, ROUND_UNNECESSARY)` would be better. But I think we don't have to worry too much about this case.